### PR TITLE
[backend] add email send retries

### LINKED ISSUE #4275: [backend] Allow retrying sending email when it fails and replace `isSent` by `emailStatus` enum which accounts for failure
## Context

Currently the notification entity has only 2 statuses regarding the email sending: `isSent: true` or `isSent: false`. When an email fails to be sent it `isSent` is set to `true` to avoid infinite retry.

## Solution

- [ ] There should be a third state something like `FAILED_TO_SEND`. So the `isSent` boolean should be renamed and changed into an enum.
- [ ] There should also be a counter to only retry sending the email a few times. These attempts shouldn't happen within the same job call. E.g if an email failed to be send during the cron job of 9:00 it will not be retried until the next cron job at maybe 9:30.
- [ ] This behavior should be included in the integration tests (to be created).

### PR DESCRIPTION
**Pull request by kdembler**
~Based on #4518, will rebase after it's merged~

Adds retries to emails that failed to be sent.

Resolves #4275

### CODE REVIEW

File: packages/server/test/notifier.test.ts [hunk 1]
```diff
@@ -459,14 +468,171 @@ describe('Notifier', () => {
...
+            }),
+          ],
+        })
+        .mockReturnValue({
+          events: [],
+          forumPostByUniqueInput: {
+            author: { handle: 'author:handle' },
+            thread: { id: 'thread:id', title: 'thread:title' },
+            text: 'Lorem Ipsum',
+          },
```
Comment (thesan) [inline packages/server/test/notifier.test.ts:582]: Is this a mistake ?
```suggestion
```

File: packages/server/src/notifier/index.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { createNotifications } from './createNotifications'
-import { sendNotifications } from './sendNotifications'
+import { processNotifications } from './sendNotifications'
```
Comment (thesan) [inline packages/server/src/notifier/index.ts:2]: Just for consistency could you either rename the function `sendNotifications` or the file `processNotifications` ?

**Review by thesan — COMMENTED** [review]
Looks great, just 2 things:

File: packages/server/test/notifier.test.ts (hunk 1 above)
Comment (kdembler) [reply packages/server/test/notifier.test.ts:582]: Not sure what you mean, but that data is needed to create the notification for `postAddedEvent`

File: packages/server/src/notifier/index.ts (hunk 2 above)
Comment (kdembler) [reply packages/server/src/notifier/index.ts]: Done 👍

File: packages/server/test/notifier.test.ts (hunk 1 above)
Comment (thesan) [reply packages/server/test/notifier.test.ts:582]: Right! It's been a while... I forgot this mocks the result of all the gql requests

**Pull request merged**